src/Cmts.ml

```diff
@@ -270,20 +270,35 @@ let partition_after_prev_or_before_next t ~prev cmts ~next =
         let same_line_as_next l =
           next.loc_start.pos_lnum = l.loc_start.pos_lnum
         in
-        let colon_before =
-          let loc_end =
-            {prev.loc_end with pos_cnum= prev.loc_end.pos_cnum - 1}
-          in
-          let char_loc = {prev with loc_start= loc_end} in
-          let str = Source.string_at t.source char_loc in
-          String.equal str ";"
+        let is_infix_id i =
```

Comment:
Copied this function for now, not sure where to put it, maybe in import/Import.ml? or creating a Utility.ml file?

Reply:
I have nothing against creating a new file. Alternatively, you could move the few functions that handle comments in `Ast.ml` to `Cmts.ml` to reverse the dependency. Both make sense to me.

Reply:
I parameterized the Ast functions with has_before/has_after functions instead, to remove the dependency to Cmts.ml